### PR TITLE
add transactionController

### PR DESCRIPTION
**Pull request by keraliss**

### CODE REVIEW

File: src/transactions/transaction.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        const result = await mockRepository.find({
+            where: { blockHeight },
+        });
+
+        expect(result).toEqual(mockTransactions);
+
+        jest.spyOn(
+            transactionsService,
+            'getTransactionByBlockHeight',
+        ).mockResolvedValue(mockTransactions);
```
Comment (theanmolsharma) [inline src/transactions/transaction.controller.spec.ts:103]: i don't think we need this. The controller result will implicitly test this behavior.

File: src/transactions/transaction.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        const result = await mockRepository.find({
+            where: { blockHash },
+        });
+        expect(result).toEqual(mockTransactions);
+
+        jest.spyOn(
+            transactionsService,
+            'getTransactionByBlockHash',
+        ).mockResolvedValue(mockTransactions);
```
Comment (theanmolsharma) [inline src/transactions/transaction.controller.spec.ts:131]: same comment here

**Review by theanmolsharma — COMMENTED** [review]
LGTM
Todos:
- address comment regarding the tests
- squash all commits into 1
- sign commits

File: src/transactions/transaction.controller.spec.ts (hunk 1 above)
Comment (keraliss) [reply src/transactions/transaction.controller.spec.ts:103]: just to clarify, you're asking me to remove this part of the test - ```
   jest.spyOn(
            transactionsService,
            'getTransactionByBlockHeight',
        ).mockResolvedValue(mockTransactions);

        const controllerResult = await controller.getTransactionByBlockHeight(
            blockHeight,
        );

        expect(
            transactionsService.getTransactionByBlockHeight,
        ).toHaveBeenCalledWith(blockHeight);
        expect(controllerResult).toEqual({ transactions: mockTransactions });
```

Comment (theanmolsharma) [reply src/transactions/transaction.controller.spec.ts:103]: no, remove the part I commented on

**Review by theanmolsharma — APPROVED** [review]

**Comment by theanmolsharma** [comment]
Closes #4

**Pull request merged**